app/src/main/java/net/gsantner/markor/activity/DocumentShareIntoFragment.java

```diff
@@ -381,54 +426,18 @@ private String formatShare(final String shared) {
             return TextUtils.join("", parts);
         }
 
-        private void showAppendDialog(int keyId) {
-            final File startFolder;
-            switch (keyId) {
-                case R.string.pref_key__favourite_files: {
-                    startFolder = GsFileBrowserListAdapter.VIRTUAL_STORAGE_FAVOURITE;
-                    break;
-                }
-                case R.string.pref_key__popular_documents: {
-                    startFolder = GsFileBrowserListAdapter.VIRTUAL_STORAGE_POPULAR;
-                    break;
-                }
-                case R.string.pref_key__recent_documents: {
-                    startFolder = GsFileBrowserListAdapter.VIRTUAL_STORAGE_RECENTS;
-                    break;
-                }
-                default: {
-                    startFolder = _appSettings.getNotebookDirectory();
-                    break;
-                }
-            }
-
-            MarkorFileBrowserFactory.showFileDialog(new GsFileBrowserOptions.SelectionListenerAdapter() {
-                @Override
-                public void onFsViewerConfig(GsFileBrowserOptions.Options dopt) {
-                    dopt.rootFolder = startFolder;
-                    dopt.newDirButtonEnable = false;
-                }
-
-                @Override
-                public void onFsViewerSelected(String request, File file, final Integer lineNumber) {
-                    appendToExistingDocumentAndClose(file, true);
-                }
-
-            }, getParentFragmentManager(), getActivity(), MarkorFileBrowserFactory.IsMimeText);
-        }
-
-
-        private void createSelectNewDocument() {
+        private void selectOrCreateDestination(final @Nullable File startFolder) {
             MarkorFileBrowserFactory.showFileDialog(new GsFileBrowserOptions.SelectionListenerAdapter() {
                 GsFileBrowserOptions.Options _dopt = null;
 
                 @Override
                 public void onFsViewerConfig(GsFileBrowserOptions.Options dopt) {
-                    dopt.rootFolder = _appSettings.getNotebookDirectory();
-                    dopt.startFolder = workingDir;
-                    dopt.okButtonText = R.string.create_new_document;
+                    dopt.rootFolder = GsFileBrowserListAdapter.VIRTUAL_STORAGE_ROOT;
```

Comment:
Switching to root as the default directory. The new organization lets us jump to notebook, favorites etc really easily